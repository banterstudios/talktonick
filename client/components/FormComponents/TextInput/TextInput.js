import React from 'react'

import PropTypes from 'prop-types'

import Input from 'client/components/Gui/Input'

import { Div } from 'glamorous'

import FormMessage from 'client/components/FormComponents/FormMessage'

const TextInput = (props) => {
  const {
    input,
    type,
    placeholder,
    tabIndex,
    meta: { touched, error, warning }
  } = props

  return (
    <Div position='relative' width='100%'>
      <Input {...input} type={type} placeholder={placeholder} tabIndex={tabIndex} />
      {touched && ((error && <FormMessage type='error'>{error}</FormMessage>) || (warning && <FormMessage type='warning'>{warning}</FormMessage>))}
    </Div>
  )
}

TextInput.propTypes = {
  type: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.any,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  onInput: PropTypes.func,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string)
  ])
}

export default TextInput
