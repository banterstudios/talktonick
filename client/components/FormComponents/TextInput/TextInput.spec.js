import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import TextInput from './TextInput'

const noop = () => {}

describe('components/FormComponents/TextInput', () => {
  let initialProps = {
    type: 'text',
    name: 'input',
    value: 'hello',
    onBlur: noop,
    onFocus: noop,
    onInput: noop,
    onChange: noop,
    disabled: false
  }

  describe('@renders', () => {
    it('in default state', () => {
      const component = mount(<TextInput { ...initialProps } />)

      expect(toJson(component)).toMatchSnapshot()
      component.unmount()
    })
  })
})
