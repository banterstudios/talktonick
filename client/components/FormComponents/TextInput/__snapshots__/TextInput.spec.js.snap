// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FormComponents/TextInput @renders in default state 1`] = `
<TextInput
  disabled={false}
  name="input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onInput={[Function]}
  type="text"
  value="hello"
>
  <glamorous.Div
    position="relative"
    width="100%"
  >
    <div
      className="css-4kgshc"
    >
      <glamorous(input)
        disabled={false}
        name="input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        type="text"
        value="hello"
      >
        <input
          className="css-1v53l60"
          disabled={false}
          name="input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onInput={[Function]}
          type="text"
          value="hello"
        />
      </glamorous(input)>
    </div>
  </glamorous.Div>
</TextInput>
`;
